src/arrays/referencing_array_and_its_elements/hard/q4/MainTest.java

```diff
@@ -0,0 +1,96 @@
+package arrays.referencing_array_and_its_elements.hard.q4;
+
+import global.BaseTest;
+import global.MethodTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.utils.ArrayUtil.arrayToInput;
+import static global.utils.ArrayUtil.generateRandomArray;
+
+public class MainTest extends BaseTest {
+    // Java
+
+    final static int n = 10;
+    final static double pivot = 88.0;
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "76.9 50.8 68.4 95.7 88.6 81.2 89.5 74.9 12.76 64.0";
+        return new Clause[]{
+                new StringLiteral("Enter " + n + " double numbers:"),
+                new NewLine(),
+                new StringLiteral("The indices of potential candidates for teaching assistants are:"),
+                new NewLine(),
+                new PlaceHolder()
+        };
+    }
+
+    public void runMain() {
+        PotentialGoodStudents.main(new String[0]);
+    }
+
+    public static String answerFor(double[] a) {
+        String ans = "";
+        for (int i = 0; i < a.length; i++) {
+            if (a[i] > pivot) {
+                ans += i + " ";
+            }
+        }
+        return ans.trim();
+    }
+
+    static Stream<Arguments> mainInputProvider() {
+        return Stream.of(
+                Arguments.of(generateRandomArray(0.0, 100, n)),
```

Comment:
You should still have different test cases here, like maybe one that starts at 88 so all of them get printed